pretty.c

```diff
@@ -0,0 +1,876 @@
+	case DRM_MODE_CONNECTOR_VIRTUAL:     return "virtual";
+	case DRM_MODE_CONNECTOR_DSI:         return "DSI";
+	case DRM_MODE_CONNECTOR_DPI:         return "DPI";
+	case DRM_MODE_CONNECTOR_WRITEBACK:   return "writeback";
+	default:                             return "unknown";
+	}
+}
+
+static const char *conn_status(drmModeConnection conn)
+{
+	switch (conn) {
+	case DRM_MODE_CONNECTED:         return "connected";
+	case DRM_MODE_DISCONNECTED:      return "disconnected";
+	case DRM_MODE_UNKNOWNCONNECTION: return "unknown";
+	default:                         return "unknown";
+	}
+}
+
+static const char *conn_subpixel(drmModeSubPixel subpixel)
+{
+	switch (subpixel) {
+	case DRM_MODE_SUBPIXEL_UNKNOWN:        return "unknown";
+	case DRM_MODE_SUBPIXEL_HORIZONTAL_RGB: return "horizontal RGB";
+	case DRM_MODE_SUBPIXEL_HORIZONTAL_BGR: return "horizontal BGR";
+	case DRM_MODE_SUBPIXEL_VERTICAL_RGB:   return "vertical RGB";
+	case DRM_MODE_SUBPIXEL_VERTICAL_BGR:   return "vertical BGR";
+	case DRM_MODE_SUBPIXEL_NONE:           return "none";
+	default:                               return "unknown";
+	}
+}
+
+static void print_connectors(struct json_object *arr)
+{
+	printf(L_VAL "Connectors\n");
+	for (size_t i = 0; i < json_object_array_length(arr); ++i) {
+		struct json_object *obj = json_object_array_get_idx(arr, i);
+		bool last = i == json_object_array_length(arr) - 1;
+
+		uint32_t id = get_object_object_uint64(obj, "id");
+		uint32_t type = get_object_object_uint64(obj, "type");
+		drmModeConnection status = get_object_object_uint64(obj, "status");
+		uint32_t phy_width = get_object_object_uint64(obj, "phy_width");
+		uint32_t phy_height = get_object_object_uint64(obj, "phy_height");
+		drmModeSubPixel subpixel = get_object_object_uint64(obj, "subpixel");
+		struct json_object *encs_arr = json_object_object_get(obj, "encoders");
+		struct json_object *modes_arr = json_object_object_get(obj, "modes");
+		struct json_object *props_obj = json_object_object_get(obj, "properties");
+
+		printf(L_LINE "%sConnector %zu\n", last ? L_LAST : L_VAL, i);
+
+		printf(L_LINE "%s" L_VAL "Object ID: %"PRIu32"\n",
+			last ? L_GAP : L_LINE, id);
+		printf(L_LINE "%s" L_VAL "Type: %s\n", last ? L_GAP : L_LINE,
+			conn_name(type));
+		printf(L_LINE "%s" L_VAL "Status: %s\n", last ? L_GAP : L_LINE,
+			conn_status(status));
+		if (status != DRM_MODE_DISCONNECTED) {
+			printf(L_LINE "%s" L_VAL "Physical size: %"PRIu32"x%"PRIu32" mm\n",
+				last ? L_GAP : L_LINE, phy_width, phy_height);
+			printf(L_LINE "%s" L_VAL "Subpixel: %s\n", last ? L_GAP : L_LINE,
+				conn_subpixel(subpixel));
+		}
+
+		bool first = true;
+		printf(L_LINE "%s" L_VAL "Encoders: {", last ? L_GAP : L_LINE);
+		for (size_t j = 0; j < json_object_array_length(encs_arr); ++j) {
+			uint32_t enc_id = get_object_uint64(
+				json_object_array_get_idx(encs_arr, j));
+			printf("%s%"PRIu32, first ? "" : ", ", enc_id);
+			first = false;
+		}
+		printf("}\n");
```

Comment:
This prints with a different format than the other { } blocks, where it's { object-id } instead of { resource-index }.
Old: `Encoders: {0}`
New: `Encoders: {55}`

I wouldn't really mind if it stayed as object-id, but I'd like it to be consistent with the other similar fields in encoders and planes.